File: 2025-12/src/day_09/boj_21921/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, x;
    cin >> n >> x;

    vector<int> v(n);
    for (int& x : v) cin >> x;

    vector<int> psum(1 + n);
    for (int i = 1; i <= n; i++) {
        psum[i] = psum[i - 1] + v[i - 1];
    }

    int mx = 0;
    int cnt = 0;
    for (int i = 0; i <= n - x; i++) {
        int diff = psum[i + x] - psum[i];

        if (diff > mx) {
            mx = diff;
            cnt = 1;
        } else if (diff == mx) {
            cnt++;
        }
    }

    if (mx == 0) {
        cout << "SAD";
    } else {
        cout << mx << '\n';
        cout << cnt << '\n';
    }
}

File: 2025-12/src/day_09/boj_21921/Main.java
package day_09.boj_21921;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int x = Integer.parseInt(st.nextToken());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int[] psum = new int[1 + n];
        for (int i = 1; i <= n; i++) {
            psum[i] = psum[i - 1] + arr[i - 1];
        }

        int max = 0;
        int cnt = 0;
        for (int i = 0; i <= n - x; i++) {
            int diff = psum[i + x] - psum[i];

            if (diff > max) {
                max = diff;
                cnt = 1;
            } else if (diff == max) {
                cnt++;
            }
        }

        if (max == 0) {
            System.out.println("SAD");
        } else {
            System.out.println(max);
            System.out.println(cnt);
        }
    }
}
